docs/upgrade.html

```diff
@@ -29,6 +29,8 @@ <h5><a id="upgrade_240_notable" href="#upgrade_240_notable">Notable changes in 2
         avoid potential misuse. The constructor <code>TopicAuthorizationException(String)</code> which was previously used for a single
         unauthorized topic was changed similarly.
     </li>
+    <li>The internal <code>PartitionAssignor</code> interface has been deprecated and replaced with a new <code>ConsumerPartitionAssignor</code> in the public API. Users
+        implementing a custom PartitionAssignor should migrate to the new interface, though an adapter has been added to temporarily support old assignor implementations. </li>
```

Comment:
Maybe no need to mention the adapter directly. Sort of sounds like we're suggesting users should rely on it directly. I think it's sufficient to say that the old interface has been deprecated and recommend users to migrate as soon as possible.